exporter/kafkaexporter/kafka_exporter_test.go

```diff
@@ -113,6 +119,106 @@ func TestTracesPusher_ctx(t *testing.T) {
 	})
 }
 
+func TestTracesPusher_attr_Kgo(t *testing.T) {
+	require.NoError(t, featuregate.GlobalRegistry().Set(franzGoClientFeatureGateName, true))
+	defer require.NoError(t, featuregate.GlobalRegistry().Set(franzGoClientFeatureGateName, false))
+
+	config := createDefaultConfig().(*Config)
+	attributeKey := "my_custom_topic_key_traces"
+	expectedTopicFromAttribute := "topic_from_traces_attr_kgo"
+	config.TopicFromAttribute = attributeKey
+
+	exp, fakeCluster := newKgoMockTracesExporter(t, *config,
+		componenttest.NewNopHost(), expectedTopicFromAttribute,
+	)
+
+	traces := testdata.GenerateTraces(1)
+	traces.ResourceSpans().At(0).Resource().Attributes().PutStr(attributeKey, expectedTopicFromAttribute)
+
+	err := exp.exportData(context.Background(), traces)
+	require.NoError(t, err)
+
+	records := fetchKgoRecords(t,
+		fakeCluster.ListenAddrs(), expectedTopicFromAttribute,
+		"test-traces-attr-group-kgo",
+		kgo.ClientID("kgo-test-consumer-traces-attr"),
```

Comment:
Is it necessary to pass a client ID in tests?

Reply:
I suppose not.